runner/src/main.rs

```diff
@@ -162,84 +162,32 @@ fn build_wasm_module(name: &str, manifest_path: &str) -> Step {
     }
 }
 
-fn is_mac_os() -> bool {
-    cfg!(macos)
-}
-
-fn bazel_build_flags() -> Vec<String> {
-    let mut flags = vec![
-        "--remote_cache=https://storage.googleapis.com/oak-bazel-cache".to_string(),
-        "--block_for_lock=false".to_string(),
-        "--show_timestamps".to_string(),
-    ];
-
-    const OAK_REMOTE_CACHE_KEY: &str = "./.oak_remote_cache_key.json";
-    if Path::new(OAK_REMOTE_CACHE_KEY).exists() {
-        flags.push(format!("--google_credentials={}", OAK_REMOTE_CACHE_KEY));
-    } else {
-        flags.push("--remote_upload_local_results=false".to_string());
-    };
-
-    flags
-}
-
-fn as_ref(v: &[String]) -> Vec<&str> {
-    v.iter().map(|x| x.as_ref()).collect()
-}
-
-fn bazel_build(extra_build_flags: &[&str], targets: &[&str]) -> Cmd {
-    let mut flags = Vec::new();
-    flags.push("build");
-
-    let b = bazel_build_flags();
-    flags.extend(as_ref(&b));
-
-    flags.extend(extra_build_flags);
-    flags.push("--");
-    flags.extend(targets);
-    Cmd::new("bazel", &flags)
-}
-
 fn build_server(opt: &BuildServer) -> Step {
-    match opt.variant.as_str() {
-        "rust" => Step::Single {
+    match opt.server_variant.as_str() {
+        "base" => Step::Single {
             name: "build rust server".to_string(),
             command: Cmd::new(
                 "cargo",
-                &[
-                    "build",
-                    "--release",
-                    "--target=x86_64-unknown-linux-musl",
-                    "--manifest-path=oak/server/rust/oak_loader/Cargo.toml",
-                ],
+                vec![
+                    match &opt.server_rust_toolchain {
+                        // This overrides the toolchain used by `rustup` to invoke the actual
+                        // `cargo` binary.
+                        // See https://github.com/rust-lang/rustup#toolchain-override-shorthand
+                        Some(server_rust_toolchain) => vec![format!("+{}", server_rust_toolchain)],
+                        None => vec![],
+                    },
+                    vec![
+                        "build".to_string(),
+                        "--release".to_string(),
+                        format!("--target={}", opt.server_rust_target),
+                        "--manifest-path=oak/server/rust/oak_loader/Cargo.toml".to_string(),
+                    ],
+                ]
+                .iter()
+                .flatten(),
```

Comment:
Personally, I'm not convinced that doing all the vec manipulation inline helps readability, but up to you.

Reply:
I agree, I have an idea for how to simplify this vastly with a macro, but haven't had time to do it yet. Maybe this weekend :)